hardware/Meteorologisk.cpp

```diff
@@ -0,0 +1,446 @@
+
+	for(int i = 0; i < (int)timeseries.size(); i++)
+	{
+		Json::Value timeserie = timeseries[i];
+		std::string stime = timeserie["time"].asString();
+
+		time_t parsedDate;
+		struct tm parsedDateTm;
+
+		if(ParseISOdatetime(parsedDate, parsedDateTm, stime))
+		{
+			double diff = difftime(mktime(&parsedDateTm), now);
+			if(diff > 0)
+			{
+				selectedTimeserie = timeserie;
+				i = timeseries.size();
+			}
+		}
+	}
+
+	if(selectedTimeserie == 0){
+		Log(LOG_ERROR,"Invalid data received, or unknown location!");
+		return;
+	}
+
+	Json::Value instantData = selectedTimeserie["data"]["instant"]["details"];
+
+	//The api already provides the temperature in celcius.
+	temperature=instantData["air_temperature"].asFloat();
+
+	if (instantData["relative_humidity"].empty()==false)
+	{
+		humidity=round(instantData["relative_humidity"].asFloat());
+	}
+	if (instantData["air_pressure_at_sea_level"].empty()==false)
+	{
+		barometric = static_cast<float>(atof(instantData["air_pressure_at_sea_level"].asString().c_str()));
+		if (barometric<1000)
+			barometric_forcast=baroForecastRain;
+		else if (barometric<1020)
+			barometric_forcast=baroForecastCloudy;
+		else if (barometric<1030)
+			barometric_forcast=baroForecastPartlyCloudy;
+		else
+			barometric_forcast=baroForecastSunny;
+
+		if (root["currently"]["icon"].empty()==false)
+		{
+			std::string forcasticon=root["currently"]["icon"].asString();
+			if ((forcasticon=="partly-cloudy-day")||(forcasticon=="partly-cloudy-night"))
+			{
+				barometric_forcast=baroForecastPartlyCloudy;
+			}
+			else if (forcasticon=="cloudy")
+			{
+				barometric_forcast=baroForecastCloudy;
+			}
+			else if ((forcasticon=="clear-day")||(forcasticon=="clear-night"))
+			{
+				barometric_forcast=baroForecastSunny;
+			}
+			else if ((forcasticon=="rain")||(forcasticon=="snow"))
+			{
+				barometric_forcast=baroForecastRain;
+			}
+		}
+	}
+
+	if (barometric!=0)
+	{
+		//Add temp+hum+baro device
+		SendTempHumBaroSensor(1, 255, temperature, humidity, static_cast<float>(barometric), barometric_forcast, "THB");
+	}
+	else if (humidity!=0)
+	{
+		//add temp+hum device
+		SendTempHumSensor(1, 255, temperature, humidity, "TempHum");
+	}
+	else
+	{
+		//add temp device
+		SendTempSensor(1, 255, temperature, "Temperature");
+	}
+
+	//Wind
+	float wind_degrees=-1;
+	float windspeed_ms=0;
+	float windgust_ms=0;
+	float wind_temp=temperature;
+	float wind_chill=temperature;
+	//int windgust=1;
+	//float windchill=-1;
+
+	if (instantData["wind_from_direction"].empty()==false)
+	{
+		wind_degrees = static_cast<float>(atof(instantData["wind_from_direction"].asString().c_str()));
+	}
+	if (instantData["wind_speed"].empty()==false)
+	{
+		windspeed_ms = static_cast<float>(atof(instantData["wind_speed"].asString().c_str()));
+	}
+	if (instantData["wind_speed_of_gust"].empty()==false)
+	{
+		windgust_ms = static_cast<float>(atof(instantData["wind_speed_of_gust"].asString().c_str()));
+	}
+	if (wind_degrees!=-1)
+	{
+		RBUF tsen;
```

Comment:
Please use CDomoticzHardwareBase::SendWind

Reply:
Done, based on the Buienrader implementation